tests/test_command.py

```diff
@@ -290,6 +305,236 @@ def test_submit_sample_size(self):
+        sys.argv = ['qds.py', 'sparkcmd', 'submit', '--cmdline', '/usr/lib/spark/bin/spark-submit --class Test Test.jar',
+                    '--name', 'test_name']
+        print_command()
+        Connection._api_call = Mock(return_value={'id': 1234})
+        qds.main()
+        Connection._api_call.assert_called_with('POST', 'commands',
+                {'macros': None,
+                 'label': None,
+                 'label_program' : None,
+                 'language' : None,
+                 'cmdline' : '/usr/lib/spark/bin/spark-submit --class Test Test.jar',
+                 'tags': None,
+                 'name': 'test_name',
+                 'arguments': None,
+                 'user_program_arguments': None,
+                 'program': None,
+                 'command_type': 'SparkCommand',
+                 'can_notify': False,
+                 'script_location': None})
+
+    def test_submit_notify(self):
+        sys.argv = ['qds.py', 'sparkcmd', 'submit', '--cmdline', '/usr/lib/spark/bin/spark-submit --class Test Test.jar',
+                    '--notify']
+        print_command()
+        Connection._api_call = Mock(return_value={'id': 1234})
+        qds.main()
+        Connection._api_call.assert_called_with('POST', 'commands',
+                {'macros': None,
+                 'label': None,
+                 'label_program' : None,
+                 'language' : None,
+                 'tags': None,
+                 'name': None,
+                 'program': None,
+                 'cmdline': '/usr/lib/spark/bin/spark-submit --class Test Test.jar',
+                 'command_type': 'SparkCommand',
+                 'arguments': None,
+                 'user_program_arguments': None,
+                 'can_notify': True,
+                 'script_location': None})
+    
+    def test_submit_python_program(self):
+        sys.argv = ['qds.py', 'sparkcmd', 'submit', '--language','python','--program', 'print "hello, world!"']
+        print_command()
+        Connection._api_call = Mock(return_value={'id': 1234})
+        qds.main()
+        Connection._api_call.assert_called_with('POST', 'commands',
+                {'macros': None,
+                 'label': None,
+                 'label_program' : None,
+                 'language' : 'python',
+                 'tags': None,
+                 'name': None,
+                 'program': "print \"hello, world!\"",
+                 'cmdline': None,
+                 'command_type': 'SparkCommand',
+                 'arguments': None,
+                 'user_program_arguments': None,
+                 'can_notify': False,
+                 'script_location': None})
+    
+    def test_submit_user_program_arguments(self):
+        sys.argv = ['qds.py', 'sparkcmd', 'submit', '--language','scala','--program',
+                    "object HelloWorld {\n\n    def main(args: Array[String]) {\n        \n        println(\"Hello, \" + args(0))\n    \n    }\n}\n",
```

Comment:
Discussed with @PraveenSeluka,
There is no point of supporting this `--program`. No one is ever going to type the complete program while invoking qds.py.

So here is what we should do:
Instead of `--program`, `--cmdline` and `--script-location`.
We should only have `--cmdline` and `--program-location`. Because we can infer the language from the program location we can get rid of the `--language` option also. This will simplify the code a lot. And will also provide a better user experience.